src/Compilers/CSharp/Portable/BoundTree/Statement.cs

```diff
@@ -445,7 +445,14 @@ internal partial class BoundCatchBlock : ICatchClause
 
         IOperation ICatchClause.Filter => this.ExceptionFilterOpt;
 
-        ILocalSymbol ICatchClause.ExceptionLocal => this.LocalOpt;
+        ILocalSymbol ICatchClause.ExceptionLocal
+        {
+            get
+            {
+                var local = this.Locals.FirstOrDefault();
+                return local?.DeclarationKind == LocalDeclarationKind.CatchVariable ? local : null;
```

Comment:
If you were to read the original comment, you would realize that this is a pre-existing condition and has nothing to do with my change. If the variable gets lifted into a closure it is no longer in the list, otherwise it stays there.